src/main/kotlin/WorkerResult.kt

```diff
@@ -1,7 +1,19 @@
 package franz
 
-enum class WorkerResult {
-    Success,
-    Retry,
-    Failure
+sealed class WorkerResult<out T> {
+    data class Success<out T>(val value: T): WorkerResult<T>()
+    object Failure: WorkerResult<Any>()
+    object Retry: WorkerResult<Any>()
+
+    fun toJobStatus(): JobStatus =
+        when(this){
+            is WorkerResult.Success<T> -> JobStatus.Incomplete
+            is WorkerResult.Retry -> JobStatus.TransientFailure
+            is WorkerResult.Failure -> JobStatus.PermanentFailure
+        }
+
+    fun value(): T? = when(this){
+        is WorkerResult.Success<T> -> this.value
+        else -> null
+    }
 }
```

Comment:
Nice! I don't know if an optional error message (`String?`) or exception (`Exception?`) would be good to have for `WorkerResult.Retry` and `WorkerResult.Failure` or if it just adds unnecessary complexity.